#! /bin/sh /usr/share/dpatch/dpatch-run

## 19_vdr-1.3.39-clre-crash-fix.dpatch by Klaus Schmidinger <Klaus.Schmidinger@cadsoft.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes segfault of vdr when using the CLRE command via SVDRP

@DPATCH@
diff -urNad vdr-1.3.39/epg.c /tmp/dpep.91NkxL/vdr-1.3.39/epg.c
--- vdr-1.3.39/epg.c	2006-01-15 14:58:30.000000000 +0100
+++ /tmp/dpep.91NkxL/vdr-1.3.39/epg.c	2006-01-20 22:16:32.753728500 +0100
@@ -935,7 +935,8 @@
   cSchedulesLock SchedulesLock(true, 1000);
   cSchedules *s = (cSchedules *)Schedules(SchedulesLock);
   if (s) {
-     s->Clear();
+     for (cSchedule *Schedule = s->First(); Schedule; Schedule = s->Next(Schedule))
+         Schedule->Cleanup(INT_MAX);
      return true;
      }
   return false;
